src/logic/credentials-storage.ts

```diff
@@ -0,0 +1,22 @@
+import { UserCredentials } from '../types/types';
+import { TokenBearer } from '../types/types';
+
+export function storeToken(token: TokenBearer) {
+  localStorage.setItem('tokenBearer', JSON.stringify(token));
+}
+
+export function retrieveToken(): TokenBearer {
+  return JSON.parse(
+    localStorage.getItem('tokenBearer') as string,
+  ) as TokenBearer;
+}
+
+export function storeUserCredentials(credentials: UserCredentials) {
+  localStorage.setItem('userCredentials', JSON.stringify(credentials));
+}
```

Comment:
The goal for storing a Record is that we can add separate entries for different URLs. Therefore I believe the implementation should probably look lile this:
```suggestion
export function storeUserCredentials(rootUrl: string, credentials: UserCredentialsResponse) {
  const allCredentialsAsString = localStorage.userCredentials;
  const allCredentials = (allCredentialsAsString ? JSON.parse(allCredentialsAsString) : {}) as UserCredentials;
  allCredentials[rootUrl] = credentials;
  localStorage.userCredentials = JSON.stringify(allCredentials);
}
```

Similar for for storeToken.